main.js

```diff
@@ -0,0 +1,138 @@
+var form = document.getElementsByClassName('post-form')[0];
+var i = 0
+
+form.addEventListener('submit', function(event1) {
+  event.preventDefault()
+  var userName = $('#name').val();
+  var post = $('#message').val();
+
+  // create a new cell in the table with the name and post content and add a quality column and a editing column
+  var newDiv = document.createElement('div');
+  newDiv.id = i
+  newDiv.className = "Rtable message"
+  $('.posts')[0].append(newDiv)
+  var postName = "<div class='Rtable-cell Rtable-cell-medium Rtable-cell-content'><span class='cell-content'>" + userName + "</span></div>"
+  var postContent = "<div class='Rtable-cell Rtable-cell-wide Rtable-cell-content'><span class='cell-content'>" + post + "</span><span><a class='comment-button' role='button'>comments</a></span><span class='comments'></span><span class='post-comment'></span></div>"
+  var postQuality = "<div id=qualCell" + i + " class='Rtable-cell Rtable-cell-content Rtable-cell-medium'><button id=upBtn" + i + " class='btn btn-xs btn-default quality-up'><span class='glyphicon glyphicon-thumbs-up'></span></button><button id=dwnBtn" + i + " class='btn btn-default btn-xs quality-down'><span class='glyphicon glyphicon-thumbs-down'></span></button><span class='q-content'><p>0</p></span></div>"
+  var postEdit = "<div class='Rtable-cell Rtable-cell-narrow Rtable-cell-empty'><button type='button' id=trash" + i +
+    " class='btn btn-default btn-xs pull-left trash'><span class='glyphicon glyphicon-trash'></span></button><button type='button' id=edit" + i +
+    " class='btn btn-default btn-xs edit'><span class='glyphicon glyphicon-pencil'></span></button></div>"
+  var tableData = postName + postContent + postQuality + postEdit
+  $('#' + i).html(tableData)
+
+  // set the inital quality attribute to zero - it can be changed by clicking thumbs up and thumbs down
+  $('#' + i).attr('qual', 0);
+  //reset the form
+  $('#message').val('')
+  $('#name').val('')
+
+  //add a comment submission form to each new post
+  $('.comment-button').parent().siblings().last().html('<input id=comment' + i + '  type="text" class="form-inline" placeholder="Comment Text"></input><input id=comment-name' + i + ' type="text" class="form-inline" placeholder="Your Name"></input><button type="button" class="btn btn-primary btn-xs	submit-comment">Post Comment</button>')
+
+  //hide the comments and submission form to start out and then toggle them on when "comments" is pressed
+  $('.comment-button').parent().nextAll().hide()
+  var hide = true
+  $('.comment-button').on('click', function() {
+    $(this).parent().nextAll().toggle(hide)
+    hide = !hide
+  })
+
+  $('.submit-comment').on('click', function(event1) {
+    //allows a user to submit a comment and provides an edit and delete icon for the comment
+    event1.stopImmediatePropagation();
+    var commentName = $(this).prev().val();
+    var commentContent = $(this).prev().prev().val()
```

Comment:
Sometimes you use semicolons and sometimes not - it's preferred to have them, but I've heard good arguments for not having them as well, just pick one.